synapse/handlers/user_directory.py

```diff
@@ -82,7 +83,20 @@ def search_users(self, user_id, search_term, limit):
                     ]
                 }
         """
-        return self.store.search_user_dir(user_id, search_term, limit)
+        results = await self.store.search_user_dir(user_id, search_term, limit)
+
+        if self.spam_checker:
+            # Remove any banned users from the results.
+            results["results"] = list(
+                filter(
+                    lambda user_info: not self.spam_checker.check_for_banned_user(
+                        user_info["user_id"], user_info["display_name"]
+                    ),
+                    results["results"],
+                )
+            )
```

Comment:
I don't usually find using `filter` vs. list comprehensions to be much clearer one way or the other. Since we're casing to a list anyway it likely makes more sense to avoid the lambda and such and just use a list comprehension. I'll update this!